libs/vectorise/include/vectorise/uint/uint.hpp

```diff
@@ -39,338 +40,1012 @@ namespace vectorise {
+
+  ///////////////////////
+  /// unary operators ///
+  ///////////////////////
+
+  constexpr UInt &operator++();
+  constexpr UInt &operator--();
+
+  //////////////////////////////
+  /// math and bit operators ///
+  //////////////////////////////
+
+  constexpr UInt  operator+(UInt const &n) const;
+  constexpr UInt  operator-(UInt const &n) const;
+  constexpr UInt  operator*(UInt const &n) const;
+  constexpr UInt  operator/(UInt const &n) const;
+  constexpr UInt  operator%(UInt const &n) const;
+  constexpr UInt  operator&(UInt const &n) const;
+  constexpr UInt  operator|(UInt const &n) const;
+  constexpr UInt  operator^(UInt const &n) const;
+  constexpr UInt &operator+=(UInt const &n);
+  constexpr UInt &operator-=(UInt const &n);
+  constexpr UInt &operator*=(UInt const &n);
+  constexpr UInt &operator/=(UInt const &n);
+  constexpr UInt &operator%=(UInt const &n);
+  constexpr UInt &operator&=(UInt const &n);
+  constexpr UInt &operator|=(UInt const &n);
+  constexpr UInt &operator^=(UInt const &n);
+  template <typename T>
+  constexpr meta::IfIsInteger<T, UInt> operator+(T const &n) const;
+  template <typename T>
+  constexpr meta::IfIsInteger<T, UInt> operator-(T const &n) const;
+  template <typename T>
+  constexpr meta::IfIsInteger<T, UInt> operator*(T const &n) const;
+  template <typename T>
+  constexpr meta::IfIsInteger<T, UInt> operator/(T const &n) const;
+  template <typename T>
+  constexpr meta::IfIsInteger<T, UInt> operator%(T const &n) const;
+  template <typename T>
+  constexpr meta::IfIsInteger<T, UInt> operator&(T const &n) const;
+  template <typename T>
+  constexpr meta::IfIsInteger<T, UInt> operator|(T const &n) const;
+  template <typename T>
+  constexpr meta::IfIsInteger<T, UInt> operator^(T const &n) const;
+  template <typename T>
+  constexpr meta::IfIsInteger<T, UInt> &operator+=(T const &n);
+  template <typename T>
+  constexpr meta::IfIsInteger<T, UInt> &operator-=(T const &n);
+  template <typename T>
+  constexpr meta::IfIsInteger<T, UInt> &operator*=(T const &n);
+  template <typename T>
+  constexpr meta::IfIsInteger<T, UInt> &operator/=(T const &n);
+  template <typename T>
+  constexpr meta::IfIsInteger<T, UInt> &operator%=(T const &n);
+  template <typename T>
+  constexpr meta::IfIsInteger<T, UInt> &operator&=(T const &n);
+  template <typename T>
+  constexpr meta::IfIsInteger<T, UInt> &operator|=(T const &n);
+  template <typename T>
+  constexpr meta::IfIsInteger<T, UInt> &operator^=(T const &n);
+
+  constexpr UInt &operator<<=(std::size_t const &n);
+  constexpr UInt &operator>>=(std::size_t const &n);
+
+  constexpr std::size_t msb() const;
+  constexpr std::size_t lsb() const;
+
+  /////////////////////////
+  /// element accessors ///
+  /////////////////////////
+
+  constexpr uint8_t   operator[](std::size_t const &n) const;
+  constexpr uint8_t & operator[](std::size_t const &n);
+  constexpr WideType  elementAt(std::size_t const &n) const;
```

Comment:
Please our standard style `ElementAt` rather than `elementAt`. When export to Etch, we `elementAt` in the Etch module definition, but not in the C++ version.

Reply:
fixed